tools/doc/versions.js

```diff
@@ -45,11 +45,16 @@ module.exports = {
       'https://raw.githubusercontent.com/nodejs/node/master/CHANGELOG.md';
     let changelog;
     const file = path.join(srcRoot, 'CHANGELOG.md');
+    const masterChangelog = path.join(srcRoot, '.master-CHANGELOG.md');
     if (kNoInternet) {
       changelog = readFileSync(file, { encoding: 'utf8' });
+    } else if (existsSync(masterChangelog)) {
```

Comment:
This should probably be using `statSync` and check whether the file is outdated – otherwise this isn’t caching, it’s just storing a file and keeping it around forever.

(Also, it’s odd that we use `fs.*Sync()` functions inside an `async` function, although it’s not that important for this script. I’d suggest using `readFile`, `stat`, `writeFile` from `fs.promises` instead, just to reflect best practices.)